### PR TITLE
🌱 Supervisor flavorgen and templates

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Add supervisor templates to flavor gen and E2E tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -336,41 +338,48 @@ generate-e2e-templates: $(KUSTOMIZE) $(addprefix generate-e2e-templates-, v1.8 v
...
+	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/dhcp-overrides" > "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/cluster-template-dhcp-overrides.yaml"
+	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/ownerrefs-finalizers" > "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/cluster-template-ownerrefs-finalizers.yaml"
 	# for IPAM tests
-	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_TEMPLATE_DIR)/main/ipam" > "$(E2E_TEMPLATE_DIR)/main/cluster-template-ipam.yaml"
+	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/ipam" > "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/cluster-template-ipam.yaml"
+	# generate supervisor templates
+	cp "$(RELEASE_DIR)/main/cluster-template-supervisor.yaml" "$(E2E_SUPERVISOR_TEMPLATE_DIR)/main/base/cluster-template-supervisor.yaml"
+	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/base" > "$(E2E_GOVMOMI_TEMPLATE_DIR)/main/cluster-template-supervisor.yaml"
```
Comment (chrischdi) [inline Makefile:368]:
```suggestion
	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_SUPERVISOR_TEMPLATE_DIR)/main/base" > "$(E2E_SUPERVISOR_TEMPLATE_DIR)/main/cluster-template-supervisor.yaml"
```
?

File: test/e2e/data/infrastructure-vsphere-supervisor/main/commons/cluster-network-CIDR.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply test/e2e/data/infrastructure-vsphere-supervisor/main/commons/cluster-network-CIDR.yaml:1]: I think we should avoid this optimization at this stage and prefer a clean directory structure.
We can eventually revisit this later when tests are working and possibly already stable

File: packaging/flavorgen/flavors/env/envsubts_consts.go [hunk 3]
```diff
@@ -18,35 +18,39 @@ limitations under the License.
...
+	VSphereNetworkVar             = "${VSPHERE_NETWORK}"
+	VSphereResourcePoolVar        = "${VSPHERE_RESOURCE_POOL}"
+	VSphereServerVar              = "${VSPHERE_SERVER}"
+	VSphereSSHAuthorizedKeysVar   = "${VSPHERE_SSH_AUTHORIZED_KEY}"
+	VSphereStoragePolicyVar       = "${VSPHERE_STORAGE_POLICY}"
+	VSphereTemplateVar            = "${VSPHERE_TEMPLATE}"
+	WorkerMachineCountVar         = "${WORKER_MACHINE_COUNT}"
+	ControlPlaneEndpointHostVar   = "${CONTROL_PLANE_ENDPOINT_IP}"
+	ControlPlaneEndpointPortVar   = "${CONTROL_PLANE_ENDPOINT_PORT:=6443}"
+	VSphereMachineClassImageVar   = "${VSPHERE_IMAGE_NAME}"
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply packaging/flavorgen/flavors/env/envsubts_consts.go:50]: Based on my understanding, the first one is an actual VM template, the other is an "image file" in a content library

**Review by chrischdi — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**